src/pages/StatisticsChart/mining/DifficultyUncleRateEpoch.tsx

```diff
@@ -15,11 +15,11 @@ const widthSpan = (value: string, currentLanguage: LanuageType) =>
 const useTooltip = () => {
   const { t } = useTranslation()
   const currentLanguage = useCurrentLanguage()
-  return ({ seriesName, data, color }: SeriesItem & { data: string }) => {
-    if (seriesName === t('block.epoch_time')) {
+  return ({ seriesName, data, color }: SeriesItem & { data?: number }) => {
+    if (seriesName === t('block.epoch_time') && typeof data === 'number') {
```

Comment:
If there is a type guard here, you can directly change the type above to `data?: unknown`.

Reply:
I would prefer a clear interface with a safe guard in runtime. Interface for developers while guard for users.

Reply:
> I would prefer a clear interface with a safe guard in runtime. Interface for developers while guard for users.

This layer of protection should be when the type is unknown. Otherwise, it is a distrust of the type, and the type should be directly set to `unknown`. Incorrect types should not continue to mislead developers.

Reply:
Indeed, how about using `!data` instead of `typeof data === 'number`, so the type of `data` is not limited to number

Reply:
I may not fully understand your meaning. If you mean that the usage below (`${data}`) accepts not only numbers but also other types, it seems that you can use `data == null`?

Reply:
> I may not fully understand your meaning. If you mean that the usage below (`${data}`) accepts not only numbers but also other types, it seems that you can use `data?: unknown` and `data != null`?

Yes, accept any type in `${data}` except empty value because `empty epoch time` is an exception

Reply:
> Yes, accept any type in `${data}` except empty value because `empty epoch time` is an exception

I think you can explicitly use `data != null && data !== ''`.

Using `!data` when the `data` type is `unknown` might be considered by future developers (including those who have forgotten) as not considering cases other than `null / undefined`, and then they might fix it to `data != null`.

Alternatively, you can add a comment explaining that !data is intentionally used here to capture the scenario of an empty string.

Reply:
Sure, I would add explanation why `data` cannot be `0`, `""`, too

Reply:
Updated by https://github.com/Magickbase/ckb-explorer-frontend/pull/139/commits/dce76b19820b2eb460b24f281f1a03ff521e2b9b